content/blog/2015-08-xx-service-discovery-etc.md

```diff
@@ -0,0 +1,318 @@
+---
+title: Custom service discovery with etcd
+created_at: 2015-06-23
+kind: article
+author_name: Fabian Reinartz 
+---
+
+In a [previous post](/blog/2015-06-01-advanced-service-discovery.md) we
+introduced numerous new ways of doing service discovery in Prometheus.
+Since then a lot has happened. We improved the internal implementation and
+received fantastic contributions from our community, adding support for
+service discovery with Kubernetes and Marathon. They will become available
+with the release of version 0.16.
+
+We also touched the topic of [custom service discovery](blog/2015-06-01-advanced-service-discovery.md#custom-service-discovery).
+
+Not every type of service discovery is generic enough to be directly included
+into the Prometheus server. Chances are your organisation has a properitary
+system in place that and you just have to make it work with Prometheus.
+This does not mean that you cannot enjoy the benefits of automatically 
+discovering new monitoring targets.
+
+In this post we will implement a small utility program that connects a custom
+service discovery approach on top of [etcd](https://coreos.com/etcd/), the
```

Comment:
Language feeling :) "a" rather than "the" because it's not *the* one key/value store.

"based on" is simply more common than "on topf of".